tests/phpunit/integration/Core/Modules/REST_Modules_ControllerTest.php

```diff
@@ -554,6 +586,100 @@ public function test_settings_rest_endpoint__post_invalid_slug() {
 		$this->assertEquals( 404, $response->get_status() );
 	}
 
+	public function test_settings_rest_endpoint__admins_with_no_view_only_settings() {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+		$this->controller->register();
+		$this->register_rest_routes();
+		$this->setup_fake_module();
+
+		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/modules/fake-module/data/settings' );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertEqualSetsWithIndex(
+			array(
+				'defaultKey' => 'default-value',
+			),
+			$response->get_data()
+		);
+	}
+
+	public function test_settings_rest_endpoint__shared_roles_with_no_view_only_settings() {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+		$this->controller->register();
+		$this->register_rest_routes();
+		$this->setup_fake_module();
+
+		$shared_with_roles = array( 'editor', 'author', 'contributor' );
+		foreach ( $shared_with_roles as $shared_with_role ) {
+			$this->share_fake_module_with_user_role( $shared_with_role );
+
+			$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/modules/fake-module/data/settings' );
+			$response = rest_get_server()->dispatch( $request );
+
+			$this->assertEquals( '200', $response->get_status() );
+			$this->assertEmpty( $response->get_data() );
+		}
+	}
+
+	public function test_settings_rest_endpoint__admins_with_view_only_settings() {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+		$this->controller->register();
+		$this->register_rest_routes();
+		$this->setup_fake_module_with_view_only_settings();
+
+		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/modules/fake-module/data/settings' );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertEqualSetsWithIndex(
+			array(
+				'defaultKey'  => 'default-value',
+				'viewOnlyKey' => 'default-value',
+			),
+			$response->get_data()
+		);
+	}
+
+	public function test_settings_rest_endpoint__unathorised_role_with_view_only_settings() {
```

Comment:
There's a typo in the name here but this test seems to be exercising the permissions, since non-admins would not be able to call this endpoint without the dashboard being shared with them, correct?

If so I think we can make that a bit more clear (something like this)
```suggestion
	public function test_settings_rest_endpoint__non_admins_require_view_only_access() {
```

Reply:
@aaemnnosttv Correct, thank you, will use that.